xonsh/aliases.py

```diff
@@ -63,6 +63,17 @@ def __init__(self, *args, **kwargs):
         self._raw = {}
         self.update(*args, **kwargs)
 
+    def register(self, func):
+        """Decorator to register the given function by name"""
+        name = func.__name__
+
+        # Strip leading underscore
+        if name.startswith("_"):
+            name = name[1:]
```

Comment:
A way to pass the alias name and do this by default will be great. Also replacing snake-case to dash-case will be good for cli usage